core/http/client/src/main/java/org/eclipse/rdf4j/http/client/QueueCursor.java

```diff
@@ -27,13 +28,13 @@
  */
 public class QueueCursor<E> extends LookAheadIteration<E, QueryEvaluationException> {
 
-	private volatile boolean done;
+	private final AtomicBoolean done = new AtomicBoolean(false);
```

Comment:
This wasn't being used in a strict sense as a hard limit with synchronized to perform once-only (consistent) actions (as evidenced by the use of queue.clear in a loop to get space for a sentinel), so switching to AtomicBoolean that has the ability to perform a lazySet that may be more useful for this scenario than a volatile variable.